### PR TITLE
[FIX] purchase_requisition: compute taxes for alternative PO creation

### PR DESCRIPTION
**Pull request by lase-odoo**
Steps to reproduce:
-
- Create a request for quotation for a product with set vendor Taxes
- Create an alternative purchase order from that RFQ with Copy Product

Current Behavior:
-
The purchase order line of the PO associated with the created alternative does not compute the taxes.

Expected behavior:
-
The taxes should be computed in the same way as if the PO was created manually and then linked to the alternatives.

Cause of the issue:
-
When creating a PO manually, the taxes of each purchase order line are computed when the product is set during the call of the `onchange_product_id` onchange method:
https://github.com/odoo/odoo/blob/64c13193ae0fe23eb1b426f5bbcd8531e9a40967/addons/purchase/models/purchase.py#L1221 https://github.com/odoo/odoo/blob/7ed0a773b6e0711b127f236ce665c35202d973bc/addons/purchase/models/purchase.py#L1055-L1061 By contrast, PO created from the `action_create_alternative` generate each `purchase.order.line` using `Command.create`. Therefore these lines will not trigger the onchange method and their taxes will not be computed.

Fix:
-
To KISS, we compute the taxes of each line just after the creation of the record using the `_compute_tax_id` method.

opw-3750719
---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/purchase_requisition/wizard/purchase_requisition_create_alternative.py [hunk 1]
```diff
@@ -60,6 +60,8 @@ def action_create_alternative(self):
             )
         vals = self._get_alternative_values()
         alt_po = self.env['purchase.order'].with_context(origin_po_id=self.origin_po_id.id, default_requisition_id=False).create(vals)
+        for line in alt_po.order_line:
+            line._compute_tax_id()
```
Comment (nle-odoo) [inline addons/purchase_requisition/wizard/purchase_requisition_create_alternative.py:64]: a detail but the method has a loop itself so I think we could do

```suggestion
        alt_po.order_line._compute_tax_id()
```

**Review by nle-odoo — APPROVED** [review]
looks good to me

it would seem better to call a higher level method, but some of the other fields like quantity seems to be already filled so seems good like this :+1:

File: addons/purchase_requisition/wizard/purchase_requisition_create_alternative.py (hunk 1 above)
Comment (lase-odoo) [reply addons/purchase_requisition/wizard/purchase_requisition_create_alternative.py]: Good catch, this has been changed accordingly !

**Comment by Whenrow** [comment]
robodoo r+
thank you

**Pull request closed without merging**